test/test_nn.py

```diff
@@ -15986,6 +15986,35 @@ def helper(input_size):
         helper([2, 3, 5, 7])
         helper([2, 3, 5, 7, 9])
 
+    # TODO: Remove onlyCPU when cuda is supported
+    @onlyCPU
+    def test_nll_loss_byte_target_matches_long(self, device):
+        N, C = 10, 4
+        data = torch.randn(N, C, device=device, requires_grad=True)
+        target = torch.empty(N, dtype=torch.long, device=device).random_(0, C)
+        gradient = torch.randn(N, device=device)
+
+        def compute_result_and_gradient(reduction, target_dtype):
+            data_clone = data.detach()
+            data_clone.requires_grad_()
+
+            prob = F.log_softmax(data_clone, dim=-1)
+            loss = nn.NLLLoss(reduction=reduction)
+            result = loss(prob, target.type(target_dtype))
+
+            if reduction == "none":
+                result.backward(gradient)
+            else:
+                result.backward(gradient[0])
```

Comment:
nit: no need for this logic or a `gradient` vector if you do `result.sum().backward()` instead- grads will still be verifiable